### DISCUSSION TITLE
Faraday Gzip

### DISCUSSION THREAD

**Discussion by bodrovis** [lostisland/faraday#1395] [Show and tell]
Hello again! So, I've started working on Faraday Gzip plugin, here's the repo https://github.com/bodrovis/faraday-gzip Everything seems to be working fine so far. I've tried it in my existing project and it works just like with faraday_middlware. However, I'm not sure I did everything correctly. For one thing, I'm not sure I should add zlib as a dependency like this: https://github.com/bodrovis/faraday-gzip/blob/master/faraday-gzip.gemspec#L36 Previously it was included like this https://github.com/lostisland/faraday_middleware/blob/704fd500c55c064a61e10e6d71e1f4cbecfa43d4/lib/faraday_middleware/gzip.rb#L16 but I'm not sure what's this `dependency` method. So, it would be cool if someone could review the code and perhaps adjust it.

All in all, I'd be happy to transfer ownership of my repo.

**Comment by olleolleolle** [comment]
The dependency method was a Faraday 1 way of postponing a require until the very last moment. A strategy to enable _optional_ dependencies. 

Now with Faraday 2, we use explicit dependencies with regular require statements, at the top of the file. Nothing fancy.

**Reply by bodrovis** [reply]
Okay, got i! So, what would be the next course of actions?

**Comment by iMacTia** [comment]
Fantastic work @bodrovis 👏!
I had a quick look and the gem looks great.

Based on the feedback I saw from the community over the past weeks, I'd only suggest to relax the `gemspec` so that it allows for Faraday 1.0 as well. This will allow people that have not upgraded to Faraday 2.0 yet to use your gem as well, instead of using the whole `faraday_middleware` 🙌.

Apart from that, I can see tests are green, so nice work 👍!

Next steps: you did the heavy lifting, it would only be right for you to release this gem on Rubygems so that everyone can start using it and take the credit for it.
We're also happy for you to keep the gem under your user, there's no need to transfer it over to us: Faraday 2.0 is all about empowering the community to contribute more and build *alongside* the core team, rather than relying on us.

Once that's done, you can advertise your new gem in our `awesome-faraday` repository. Just open a PR to add it to the README and I'll get it merged 🎉

**Reply by bodrovis** [reply]
Great, thank you so much for the feedback @iMacTia I'll work on it in the next few days 👍
